adapters/httpapi_compact.c

```diff
@@ -227,6 +236,20 @@ HTTP_HANDLE HTTPAPI_CreateConnection(const char* hostName)
             tlsio_config.underlying_io_interface = NULL;
             tlsio_config.underlying_io_parameters = NULL;
 
+            HTTP_PROXY_IO_CONFIG proxy_config;
+            if (proxyHost != NULL && strlen(proxyHost) > 0)
+            {
+                proxy_config.hostname = hostName;
+                proxy_config.port = 443;
+                proxy_config.proxy_hostname = proxyHost;
+                proxy_config.proxy_port = proxyPort;
+                proxy_config.username = proxyUsername;
+                proxy_config.password = proxyPassword;
+
+                tlsio_config.underlying_io_interface = http_proxy_io_get_interface_description();
```

Comment:
what happen if http_proxy_io_get_interface_description() returns null? you probably also need to clean up resources allocated so far. This is unfortunately C code...

Reply:
Made some refinement.
If http_proxy_io_get_interface_description() returns null, the code will continue execution, as if the proxy is not set, and there is no extra resources that need be released

Reply:
I think you should return error if cannot get proxy interface. Otherwise, it is unclear for users why proxy does not work as expected.

Reply:
OK, updated based on your suggestion